build/Dockerfile-debug

```diff
@@ -0,0 +1,49 @@
+FROM philipgough/go-centos:1.10.1 AS build-env
```

Comment:
Is there a library image we can use here?

Reply:
My initial thought with this is that it should mirror the existing build images as best as possible. Even better if you could modify the `Dockerfile-canary` with build arguments to optionally build it for debugging.

Reply:
This is the first time I've seen a Dockerfile like this :+1:

Reply:
> Is there a library image we can use here?

Couldn't find one

> Even better if you could modify the Dockerfile-canary with build arguments to optionally build it for debugging.

That could be done but not without the modified entrypoint script right?

Reply:
The main problem here is what we would need to have go installed in the broker base, which is currently just based off centos to achieve this, where as at the moment we build the binaries and copy them in. I think anything other than a build container is problematic if that makes sense?

Reply:
My primary reason for starting this conversation is, if we can avoid having another Dockerfile, that would best. Both the `Dockerfile-latest` and `Dockerfile-nightly` use rpms so we can't do anything there. `Dockerfile-localdev` does the copying but `Dockerfile-canary`, as you can see [here](https://github.com/openshift/ansible-service-broker/blob/master/build/Dockerfile-canary#L44) builds the broker from source.

I would be curious: 

1. how many contributors are using the localdev variant, if that's even well supported anymore I don't know.
1. if we could simply build `dlv` in the `Dockerfile-canary` and update the broker-apb to deploy the broker with the appropriate `command` and `args` in the deployment(config) for debugging.

Reply:
@djzager @PhilipGough I use the `Dockerfile-localdev` ALL THE TIME :) my workflow recently has been to do a `make build-image ORG=jmrodri TAG=demo` and pushing that to my org, then redeploying with catasb. I would be okay movign the debug stuff to the localdev dockerfile